### PR TITLE
Story/FLCRM-8689 Add record key fields to records

### PR DESCRIPTION
**Pull request by toddrun**
**Proposed Changes:**

Adds `record_key` and `record_sequence` to records
Adds `record_record_key` and `record_record_sequence` to repeatable records

**Testing Steps:**

Run `make test`

**Caveats, conditions and confessions:**

There is an assumption by the author that `record_record_key` and `record_record_sequence` can/will be populated with the `record_key` and `record_sequence` from the parent record, into each repeatable record. @zhm, please confirm this understanding is accurate, and/or that this addition seems reasonable. Though we're not targeting Forms with Repeatables initially, putting in a little extra thought/effort now may pay off later (versus having a whole in functionality which someone has to try to uncover and correct).

### CODE REVIEW

File: dist/fulcrum-schema.min.js [hunk 1]
```diff
@@ -3120,6 +3126,12 @@ Schema.systemRepeatableTableColumns = [{
 }, {
   name: 'parent_resource_id',
   type: 'string'
+}, {
+  name: 'record_record_key',
+  type: 'string'
+}, {
+  name: 'record_record_sequence',
+  type: 'integer'
```
Comment (toddrun) [inline dist/fulcrum-schema.min.js]: @jahndis made a good point: maybe these should be "parent_record_xxx"?

The idea is that repeatables would have a copy of the parent/root records key and sequence. Not sure whether that's what the code is actually doing, but that's the intent.

Comment (toddrun) [reply dist/fulcrum-schema.min.js]: So, it turns out "record" == "root" while "parent" == "one level up". For example, "id" is the current records id, while "parent_record_id" is the immediate parents id and "record_id" is the root record id, which could be grand or great-grand parent. Based on that, "record_record_key" is probably the correct term.

File: test/fixtures/postgres-create-form.sql [hunk 2]
```diff
@@ -441,7 +453,6 @@ CREATE INDEX idx_form_67777_changeset_resource_id ON organization_1.form_67777 U
 CREATE INDEX idx_form_67777_values_record_resource_id ON organization_1.form_67777_values USING btree (record_resource_id);
 CREATE INDEX idx_form_67777_values_parent_resource_id ON organization_1.form_67777_values USING btree (parent_resource_id);
 CREATE INDEX idx_form_67777_values_text_value ON organization_1.form_67777_values USING btree (text_value);
-CREATE INDEX idx_form_67777_values_key ON organization_1.form_67777_values USING btree (key);
```
Comment (jahndis) [inline test/fixtures/postgres-create-form.sql:444]: why was this removed? (asking completely out of ignorance and that it looked suspicious)

File: src/schemas/v2.js [hunk 3]
```diff
@@ -1803,15 +1819,15 @@ Schema.systemRepeatableTableIndexes = [
...
+  { columns: [ 'changeset_resource_id' ], method: 'btree' },
+  { columns: [ 'record_record_sequence' ], method: 'btree' }
 ];
 
 Schema.systemValuesTableIndexes = [
   // { columns: [ 'record_id' ], method: 'btree' },
   { columns: [ 'record_resource_id' ], method: 'btree' },
   { columns: [ 'parent_resource_id' ], method: 'btree' },
-  { columns: [ 'text_value' ], method: 'btree' },
-  { columns: [ 'key' ], method: 'btree' }
```
Comment (jahndis) [inline src/schemas/v2.js:1814]: was this supposed to be removed? (same ignorance)

**Review by jahndis — COMMENTED** [review]
I also saw references to `project_id` in the following files:
test/fixtures/sqlite-create-form.sql
test/fixtures/sqlite-update-form.sql
src/schemas/v1.js
sql/template.sql
Do we also need to update those?

**Comment by toddrun** [comment]
This is going to be completed in a different ticket, following a spike. At this point, this PR holds no value
